torch/_dynamo/symbolic_convert.py

```diff
@@ -125,6 +126,41 @@ def generic_jump(truth_fn: typing.Callable, push: bool):
     def inner(self: "InstructionTranslatorBase", inst: Instruction):
         value: VariableTracker = self.pop()
         self.output.guards.update(value.guards)
+        if config.rewrite_assert_with_torch_assert:
+            # detect if this jump instruction is assert
+            # Python assertion is in following format:
+            # 18 POP_JUMP_IF_TRUE       24
+            # 20 LOAD_ASSERTION_ERROR
+            # 22 RAISE_VARARGS           1
+            #
+            # We find this sequence and rewrite them as torch._assert
+
+            # DETECT POP_JUMP_IF_TRUE
+            if truth_fn is operator.truth and not push:
+                current_instruction_pointer = self.instruction_pointer
+                inst = self.instructions[current_instruction_pointer]
+                # Detect LOAD_ASSERTION_ERROR
+                if inst.opname == "LOAD_ASSERTION_ERROR":
+                    current_instruction_pointer += 1
+                    if current_instruction_pointer < len(self.instructions):
+                        inst_for_raise = self.instructions[current_instruction_pointer]
+                        # DETECT RAISE_VARARGS
+                        if inst_for_raise.opname == "RAISE_VARARGS":
```

Comment:
If this were a helper function, you could return rather than deeply nesting if statements.

Reply:
done